tests/cupy_tests/manipulation_tests/test_add_remove.py

```diff
@@ -29,3 +32,58 @@ def test_unique_inverse(self, xp, dtype):
     def test_unique_counts(self, xp, dtype):
         a = testing.shaped_random((100, 100), xp, dtype)
         return xp.unique(a, return_counts=True)[1]
+
+
+@testing.parameterize(*testing.product({
+    'trim': ['fb', 'f', 'b']
+}))
+@testing.gpu
+class TestTrim_zeros(unittest.TestCase):
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_array_equal()
+    def test_trim_non_zeros(self, xp, dtype):
+        a = xp.array([-1, 2, -3, 7], dtype=dtype)
+        return xp.trim_zeros(a, trim=self.trim)
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_array_equal()
+    def test_trim_trimmed(self, xp, dtype):
+        a = xp.array([1, 0, 2, 3, 0, 5], dtype=dtype)
+        return xp.trim_zeros(a, trim=self.trim)
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_array_equal()
+    def test_trim_all_zeros(self, xp, dtype):
+        a = xp.zeros(shape=(1000,), dtype=dtype)
+        return xp.trim_zeros(a, trim=self.trim)
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_array_equal()
+    def test_trim_front_zeros(self, xp, dtype):
+        a = xp.array([0, 0, 4, 1, 0, 2, 3, 0, 5], dtype=dtype)
+        return xp.trim_zeros(a, trim=self.trim)
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_array_equal()
+    def test_trim_back_zeros(self, xp, dtype):
+        a = xp.array([1, 0, 2, 3, 0, 5, 0, 0, 0], dtype=dtype)
+        return xp.trim_zeros(a, trim=self.trim)
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_array_equal()
+    def test_trim_empty(self, xp, dtype):
+        a = testing.empty(xp, dtype)
+        return xp.trim_zeros(a, trim=self.trim)
+
+    @testing.for_all_dtypes()
+    def test_trim_zero_dim(self, dtype):
+        a = testing.shaped_arange((), cupy, dtype)
+        with pytest.raises(TypeError):
+            cupy.trim_zeros(a, trim=self.trim)
```

Comment:
Please check both NumPy and CuPy raises the error of the same type like as follows:
```py
for xp in (numpy, cupy):
    a = testing.shaped_arange(...)
    with pytest.raises(TypeError):
        ....
```